rpc/core/types/responses.go

```diff
@@ -272,3 +272,8 @@ type ResultEvent struct {
 	Data   types.TMEventData   `json:"data"`
 	Events map[string][]string `json:"events"`
 }
+
+// ResultShareProof API proof response of a set of shares
+type ResultShareProof struct {
+	Proof types.ShareProof `json:"proof"`
```

Comment:
[question] why not `ShareProof`?

```suggestion
	ShareProof types.ShareProof `json:"share_proof"`
```

Reply:
no reason, if you prefer `share_proof` I can switch

Reply:
Since this `ShareProof` is actually composed of a bunch of other proofs, I feel like `ShareProof` helps differentiate it from the several other types of proof inside. I don't have a strong preference so defer to you.

Reply:
[7252ab8](https://github.com/celestiaorg/celestia-core/pull/1351/commits/7252ab8d2af2ae652d814bb0776cda1f499f70ed)